src/System/Wlog/Handler.hs

```diff
@@ -54,6 +54,9 @@ class LogHandler a where
     -- | Gets the current level.
     getLevel :: a -> Severity
 
+    -- | Needed for printing 'Error' only to @stderr@
+    shouldPrintError :: a -> Bool
```

Comment:
Approach with `shouldPrintError` is quite hacky and dirty... I think you can agree with this. And it's not quite obvious which solution to use. I have several proposals:

1. We can change `getLevel` function to return `Set Severity` for this `LogHandler`. This approach allows us to specify maximum flexibility for defining which severity should this logger use. For example, currently it's not possible to log only `Debug` messages with some logger. But with this approach it should be possible. Though, we might think more about changing configuration... It becomes really strange that our functions work with `Set` but in `yaml` files for our logger config we specify only one severity.
2. Less flexible approach is to define range of `Severity` instead of `Set`. Like `lowestSeverity :: a -> Severity` and `greatestSeverity :: a -> Severity`. This approach has less power than the previous one. But it's easier to implement and it's easier to migrate from previous config.
3. Instead of modifying `getLevel` function we can add method `excludedSeverities :: a -> Set Severity`. So it's more like your `shouldPrintError` but generalized to allow you to exclude any severity you want. This plays well with current notion of _lowestLevelOfSeverity_ and doesn't require you to change config format in backwards incompatible way.
4. Maybe something else.

What do you think @vrom911? Which one do you like more? Or what other ideas do you have?

Reply:
I'm voting for the first one. As you said it's the most flexible and configurable for any needs :100:

Reply:
Okay, so let's go with 1st option :1st_place_medal: